src/modules/settings/components/Table.jsx

```diff
@@ -0,0 +1,242 @@
+import React, {useRef, useEffect, useContext} from 'react';
+
+import Table from 'rsuite/lib/Table/index.js';
+import Button from 'rsuite/lib/Button/index.js';
+import Icon from 'rsuite/lib/Icon/index.js';
+import Dropdown from 'rsuite/lib/Dropdown/index.js';
+import Popover from 'rsuite/lib/Popover/index.js';
+import Whisper from 'rsuite/lib/Whisper/index.js';
+import Placeholder from 'rsuite/lib/Placeholder/index.js';
+import minus from '../../../assets/icons/minus-square-solid.svg';
+import ModalContext from '../context.js';
+
+const {Column, HeaderCell, Cell} = Table;
+
+const Menu = ({onSelect, options}) => (
+  <Dropdown.Menu onSelect={onSelect}>
+    {options.map((option, index) => (
+      <Dropdown.Item key={option} eventKey={index}>
+        {option}
+      </Dropdown.Item>
+    ))}
+  </Dropdown.Menu>
+);
+
+const MenuPopover = ({options, onSelect, ...rest}) => (
+  <Popover {...rest} full>
+    <Menu options={options} onSelect={onSelect} />
+  </Popover>
+);
+
+class CustomWhisper extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleSelectMenu = this.handleSelectMenu.bind(this);
+  }
+
+  handleSelectMenu(eventKey) {
+    const {onChange, rowData, dataKey} = this.props;
+    onChange(rowData.id, dataKey, eventKey);
+    this.trigger.hide();
+  }
+
+  render() {
+    const {children, options} = this.props;
+    return (
+      <Whisper
+        placement="autoVerticalStart"
+        trigger="click"
+        triggerRef={(ref) => {
+          this.trigger = ref;
+        }}
+        speaker={<MenuPopover options={options} onSelect={this.handleSelectMenu} />}>
+        {children}
+      </Whisper>
+    );
+  }
+}
+
+function useOutsideAlerter(ref, onClick, rowData) {
+  useEffect(() => {
+    function handleClickOutside(event) {
+      if (ref.current && !ref.current.contains(event.target)) {
+        onClick('edit', rowData.id);
```

Comment:
`edit` doesn't look used. why do we pass this?

Reply:
A remnant from a previous version where I had 2 buttons in 1 cell